eZ/Publish/Core/Helper/TranslationHelper.php

```diff
@@ -69,7 +69,7 @@ public function getTranslatedContentName(Content $content, $forcedLanguage = nul
      *
      * @return string
      */
-    private function getTranslatedContentNameByVersionInfo(VersionInfo $versionInfo, $forcedLanguage = null)
+    public function getTranslatedContentNameByVersionInfo(VersionInfo $versionInfo, $forcedLanguage = null)
```

Comment:
@alongosz this one got me confused. Shouldn't we keep those old type-hints as a part of BC policy? I mean, this method is already used in the `getTranslatedContentName` method.

Reply:
> @alongosz this one got me confused. Shouldn't we keep those old type-hints as a part of BC policy, even if method itself was private? I mean, this method is already used in the `getTranslatedContentName` method and the `getTranslatedContentName` doesn't have strict type-hints.

@barw4 It always depends on a specific case. If the method was private, there's nothing wrong with changing type hints as long as other methods which are invoking it are ready for that. You can even completely get rid of that method and replace it with something else. If still it's doubtful (because of e.g. the usage in `getTranslatedContentName`), we need to look at that usage and consequences of the change. If anyone passes anything else there, then it's a serious issue that can be detected thanks to the change. The code won't work if you pass something that isn't "stringable" as both an argument and `VersionInfo::getName` result (meaning it would've already crashed for someone doing that).

Someone could still argue that wrong input shouldn't result in a crash [1]. Fortunately we're in not any danger here. The thing is - this class doesn't have strict types declared. That means that even if we passed or returned e.g. `int(1)` it still would've been cast to `string('1')`.

Last but not least, we need to assess overall risks and impact to be able gradually reduce technical debt. In this case those are minimal and it's possible that at some point the entire class will get an overhaul, within reason. Now, that the method has become, public we're gonna need to be more careful about any changes to its signature, so right now is the best moment to do it.

-----------
[1] 
> On two occasions I have been asked, - "Pray, Mr. Babbage, if you put into the machine wrong figures, will the right answers come out?"
I am not able rightly to apprehend the kind of confusion of ideas that could provoke such a question

~ Charles Babbage